File: MyBudget-back/src/main/java/com/example/mybudget/models/requests/TransactionRequest.java
package com.example.mybudget.models.requests;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class TransactionRequest {
    @NotBlank
    private String description;
    @NotNull
    private BigDecimal amount;
    @NotBlank
    private String currency;
    @NotNull
    private Integer accountId;
}

File: MyBudget-front/src/app/models/transaction-request.ts
export class TransactionRequest {
  description: string;
  amount: number;
  currency: string;
  accountId: number;

  constructor(
    description: string,
    amount: number,
    currency: string,
    accountId: number
  ) {
    this.description = description;
    this.amount = amount;
    this.currency = currency;
    this.accountId = accountId;
  }
}
